File: src/main/scala/com/study/interviewbit/strings/Strstr.java
package com.study.interviewbit.strings;
/**
Implement strStr().

        strstr - locate a substring ( needle ) in a string ( haystack ).
 */
public class Strstr {
    public int strStr(final String haystack, final String needle) {
        if ((needle.isEmpty()) || (haystack.isEmpty())){
            return -1;
        }
        return haystack.indexOf(needle);
    }
}

File: src/main/scala/com/study/interviewbit/amazon/leet/strings/Strstr.scala
package com.study.interviewbit.amazon.leet.strings

/**
 *
 * @author andrii_lytvynskyi
 */
object Strstr {
  def strStr(haystack: String, needle: String): Int = {
    for (start <- 0 until (haystack.length - needle.length + 1)) {
      if (haystack.substring(start, start + needle.length) == needle){
        return start
      }
    }
    return -1
  }

}
